onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -1662,13 +1662,32 @@ def aten_chunk(self: TTensor, chunks: int, dim: int = 0) -> Sequence[TTensor]:
     return op.SplitToSequence(self, list_split, axis=dim)
 
 
-@torch_op(("aten::clamp", "aten::clamp.Tensor"), trace_only=True)
-def aten_clamp(self: TReal, min: Optional[TReal] = None, max: Optional[TReal] = None) -> TReal:
-    """clamp(Tensor self, Tensor? min=None, Tensor? max=None) -> Tensor"""
-    clamped = self
+@torch_op("aten::clamp", trace_only=True)
+def aten_clamp(self: TReal, min: Optional[float] = None, max: Optional[float] = None) -> TReal:
+    """clamp(Tensor self, Scalar? min=None, Scalar? max=None) -> Tensor"""
 
     if min is None and max is None:
-        return clamped
+        return op.Identity(self)
+
+    if min is not None:
+        min = op.CastLike(min, self)
+
+    if max is not None:
+        max = op.CastLike(max, self)
+
+    return op.Clip(self, min, max)
```

Comment:
Can min, max (dtype: float) be input of Clip? I see Clip needs T on min and max.

Reply:
There is a CastLike above for this